docs/SECURITY.md

```diff
@@ -0,0 +1,277 @@
+to deal with DDoS attacks.
+
+**Potential additional defense mechanism**: Assuming the last criteria is not
+met (it seems to be quite difficult to put in practice), there still needs to be
+at least one drand nodes that is not under attack to aggregate the partial
+beacons. To increase the chance of reconstructing the final random beacon from
+the partial beacons that "leaks" out from drand nodes under attack, it could
+also be possible to setup aggregator nodes. Such nodes could be under heavy
+protection, potentially with a more centralized governance, whose job is only
+collects the different partial beacons and aggregates them to deliver them to
+the relay network. There could be many such aggregators nodes such that the
+chance of getting at least one of these received enough threshold beacon
+drastically increases.
+
+#### Corruption of the drand network
+
+**Scenario #1**: Corruption of less than threshold of nodes
+
+In this scenario, the attacker corrupts _less_ than a threshold of drand nodes.
+_Consequence_: The attacker is _not_ able to derive any meaningful
+information with respect to beacon chain (i.e. he can't derive future beacons).
+However, it is assumed it has now access to the long term private key of each
+compromised node.
+
+**Scenario #2**: Corruption of more than a threshold of nodes
+
+In this scenario, the threat model of drand is now violated and thus is the
+scenario to avoid at all costs: the attacker corrupts _at least_ a threshold of
+drand nodes.  _Consequence_: The attacker is now able to derive the whole chain,
+i.e. it can derive any given random beacon of the chain. The drand randomness is
+not _unpredictable_ anymore from the point of view of the attacker.  However,
+the drand randomness stays _unbiasable_: attacker is not able to change the
+randomness in any way.
+
+**Mitigation**: Proactive resharing allows both to 
+1. Let a new group of nodes take over the randomness generation, potentially
+   with more nodes and higher threshold. 
+2. Refresh shares for nodes that participate in the resharing in the new group:
+   partial beacons created from an old share is not validated by the members of
+   the new group.
+
+Given a "periodical" resharing with more nodes, it makes it harder for the
+attacker to maintain the grasp on the shares of the drand nodes since he must
+have continuous control over the drand node itself. If the operator of a
+corrupted drand node recovers from the attacker's intrusion, after a resharing,
+the attacker's share is invalid. Moreover, a resharing with more nodes highers
+the bar for the attacker to attain the second scenario both because of the
+previous argument and because attacker needs now to corrupts more nodes than in
+the initial group now.
+
+As such, it is recommended to reshare often, _even if_ between the same nodes,
+as it creates new shares.
+
+### Distributed Key Generation
+
+#### DDoS attacks
```

Comment:
this is mitigated by the fact that at the end of the DKG process all nodes know which subset completed the process, and could potentially decide to retry if they aren't happy with that result?

Reply:
So the DKG fails is there are less than a threshold of nodes stated as "QUALIFIED". If it's not the case, then yes, if some nodes gets dropped, operators can simply re-run the DKG  or re-reshare to retry.
I'll add more details around that.